presto-native-execution/src/test/java/com/facebook/presto/nativeworker/TestPrestoNativeArrayOperators.java

```diff
@@ -0,0 +1,1491 @@
+        assertQuery("select ARRAY [1.0, 2.0, 3.0] != ARRAY [1.0, 2.0, 3.0]");
+        assertQuery("select ARRAY [1.0, 2.0, 3.0] != ARRAY [1.0, 2.0, 3.1]");
+        assertQuery("select ARRAY [1234567890.1234567890, 9876543210.9876543210, 123123123456.6549876543] " +
+                "!= ARRAY [1234567890.1234567890, 9876543210.9876543210, 123123123456.6549876543]");
+        assertQuery("select ARRAY [1234567890.1234567890, 9876543210.9876543210, 123123123456.6549876543] " +
+                "!= ARRAY [1234567890.1234567890, 9876543210.9876543210, 0]");
+        assertQuery("select ARRAY [1, 2, null] = ARRAY [1, null]");
+        assertQuery("select ARRAY ['1', '2', null] = ARRAY ['1', null]");
+        assertQuery("select ARRAY [1.0, 2.0, null] = ARRAY [1.0, null]");
+        assertQuery("select ARRAY [1.0E0, 2.0E0, null] = ARRAY [1.0E0, null]");
+        assertQuery("select ARRAY [1, 2, null] = ARRAY [1, 2, null]");
+        assertQuery("select ARRAY ['1', '2', null] = ARRAY ['1', '2', null]");
+        assertQuery("select ARRAY [1.0, 2.0, null] = ARRAY [1.0, 2.0, null]");
+        assertQuery("select ARRAY [1.0E0, 2.0E0, null] = ARRAY [1.0E0, 2.0E0, null]");
+        assertQuery("select ARRAY [1, 3, null] = ARRAY [1, 2, null]");
+        assertQuery("select ARRAY [1E0, 3E0, null] = ARRAY [1E0, 2E0, null]");
+        assertQuery("select ARRAY ['1', '3', null] = ARRAY ['1', '2', null]");
+        assertQuery("select ARRAY [ARRAY[1], ARRAY[null], ARRAY[2]] = ARRAY [ARRAY[1], ARRAY[2], ARRAY[3]]");
+        assertQuery("select ARRAY [ARRAY[1], ARRAY[null], ARRAY[3]] = ARRAY [ARRAY[1], ARRAY[2], ARRAY[3]]");
+
+        assertQuery("select ARRAY [10, 20, 30] != ARRAY [5]");
+        assertQuery("select ARRAY [10, 20, 30] = ARRAY [5]");
+        assertQuery("select ARRAY [1, 2, 3] != ARRAY [3, 2, 1]");
+        assertQuery("select ARRAY [1, 2, 3] = ARRAY [3, 2, 1]");
+        assertQuery("select ARRAY [TRUE, FALSE, TRUE] != ARRAY [TRUE]");
+        assertQuery("select ARRAY [TRUE, FALSE, TRUE] = ARRAY [TRUE]");
+        assertQuery("select ARRAY [TRUE, FALSE] != ARRAY [FALSE, FALSE]");
+        assertQuery("select ARRAY [TRUE, FALSE] = ARRAY [FALSE, FALSE]");
+        assertQuery("select ARRAY [1.1E0, 2.2E0, 3.3E0, 4.4E0] != ARRAY [1.1E0, 2.2E0]");
+        assertQuery("select ARRAY [1.1E0, 2.2E0, 3.3E0, 4.4E0] = ARRAY [1.1E0, 2.2E0]");
+        assertQuery("select ARRAY [1.1E0, 2.2E0, 3.3E0] != ARRAY [11.1E0, 22.1E0, 1.1E0, 44.1E0]");
+        assertQuery("select ARRAY [1.1E0, 2.2E0, 3.3E0] = ARRAY [11.1E0, 22.1E0, 1.1E0, 44.1E0]");
+        assertQuery("select ARRAY ['puppies', 'kittens', 'lizards'] != ARRAY ['puppies', 'kittens']");
+        assertQuery("select ARRAY ['puppies', 'kittens', 'lizards'] = ARRAY ['puppies', 'kittens']");
+        assertQuery("select ARRAY ['puppies', 'kittens'] != ARRAY ['z', 'f', 's', 'd', 'g']");
+        assertQuery("select ARRAY ['puppies', 'kittens'] = ARRAY ['z', 'f', 's', 'd', 'g']");
```

Comment:
@mknegi I don't think these are using presto cpp worker. It is probably getting evaluated in the co-ordinator.
Would you plesae replace the constants with `select m != n from value(....);`

Reply:
@aditi-pandit This is the query plan, only the final result (which is bool) for these queries is seems to be projected. CMIW.

```
presto> explain analyze  select ARRAY ['puppies', 'kittens', 'lizards'] != ARRAY ['puppies', 'kittens'];
                                                        Query Plan                                                         
---------------------------------------------------------------------------------------------------------------------------
 Fragment 1 [SINGLE]                                                                                                       
     CPU: 1.29ms, Scheduled: 1.53ms, Input: 0 rows (0B); per task: avg.: 0.00 std.dev.: 0.00, Output: 1 row (45B), 1 tasks 
     Output layout: [expr]                                                                                                 
     Output partitioning: SINGLE []                                                                                        
     Stage Execution Strategy: UNGROUPED_EXECUTION                                                                         
     - Project[PlanNodeId 1][projectLocality = LOCAL] => [expr:boolean]                                                    
             CPU: 0.00ns (?%), Scheduled: 1.00ms (100.00%), Output: 1 row (45B)                                            
             Input avg.: 0.06 rows, Input std.dev.: 387.30%                                                                
             expr := BOOLEAN'true'                                                                                         
         - LocalExchange[PlanNodeId 240][ROUND_ROBIN] () => []                                                             
                 CPU: 0.00ns (?%), Scheduled: 0.00ns (0.00%), Output: 1 row (0B)                                           
                 Input avg.: 1.00 rows, Input std.dev.: 0.00%                                                              
             - Values[PlanNodeId 0] => []                                                                                  
                     CPU: 0.00ns (?%), Scheduled: 0.00ns (0.00%), Output: 1 row (0B)                                       
                     Input avg.: 0.00 rows, Input std.dev.: ?%                                                             
                     ()                                                                                                    
                                                                                                                           
                                                                                                                           
(1 row)

Query 20240815_052031_00007_v55mn, FINISHED, 2 nodes
Splits: 34 total, 34 done (100.00%)
[Latency: client-side: 0:02, server-side: 0:02] [0 rows, 0B] [0 rows/s, 0B/s]

presto> explain analyze select ARRAY ['puppies', 'kittens'] = ARRAY ['z', 'f', 's', 'd', 'g'];
                                                        Query Plan                                                         
---------------------------------------------------------------------------------------------------------------------------
 Fragment 1 [SINGLE]                                                                                                       
     CPU: 1.39ms, Scheduled: 5.32ms, Input: 0 rows (0B); per task: avg.: 0.00 std.dev.: 0.00, Output: 1 row (45B), 1 tasks 
     Output layout: [expr]                                                                                                 
     Output partitioning: SINGLE []                                                                                        
     Stage Execution Strategy: UNGROUPED_EXECUTION                                                                         
     - Project[PlanNodeId 1][projectLocality = LOCAL] => [expr:boolean]                                                    
             CPU: 1.00ms (100.00%), Scheduled: 4.00ms (100.00%), Output: 1 row (45B)                                       
             Input avg.: 0.06 rows, Input std.dev.: 387.30%                                                                
             expr := BOOLEAN'false'                                                                                        
         - LocalExchange[PlanNodeId 240][ROUND_ROBIN] () => []                                                             
                 CPU: 0.00ns (0.00%), Scheduled: 0.00ns (0.00%), Output: 1 row (0B)                                        
                 Input avg.: 1.00 rows, Input std.dev.: 0.00%                                                              
             - Values[PlanNodeId 0] => []                                                                                  
                     CPU: 0.00ns (0.00%), Scheduled: 0.00ns (0.00%), Output: 1 row (0B)                                    
                     Input avg.: 0.00 rows, Input std.dev.: ?%                                                             
                     ()                                                                                                    
                                                                                                                           
                                                                                                                           
(1 row)

```